### ISSUE TITLE
[REVIEW]: E2EDNA 2.0: Python Pipeline for Simulating DNA Aptamers with Ligands

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#4182]
**Submitting author:** @taoliu032 (<a href="http://orcid.org/0000-0002-1082-5570">Tao Liu</a>)
**Repository:** https://github.com/siminegroup/E2EDNA2
**Branch with paper.md** (empty if default branch): JOSS
**Version:** v2.0.0
**Editor:** @csoneson
**Reviewers:** @JoaoRodrigues, @rvhonorato
**Archive:** 10.5281/zenodo.6546661

 

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/313422a651f2ff1ad183e898f4f5f5a3/status.svg)](https://joss.theoj.org/papers/313422a651f2ff1ad183e898f4f5f5a3)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/313422a651f2ff1ad183e898f4f5f5a3"><img src="https://joss.theoj.org/papers/313422a651f2ff1ad183e898f4f5f5a3/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/313422a651f2ff1ad183e898f4f5f5a3/status.svg)](https://joss.theoj.org/papers/313422a651f2ff1ad183e898f4f5f5a3)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/siminegroup/E2EDNA2" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@JoaoRodrigues & @rvhonorato, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @csoneson know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

## Review checklist for @JoaoRodrigues
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/siminegroup/E2EDNA2">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/siminegroup/E2EDNA2">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@taoliu032) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

## Review checklist for @rvhonorato
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/siminegroup/E2EDNA2">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/siminegroup/E2EDNA2">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@taoliu032) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @JoaoRodrigues, @rvhonorato it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.20 s (288.0 files/s, 81482.6 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
XML                              9              0              0           7642
Python                          36           1410           1616           4901
Markdown                         8             70              0            182
YAML                             1              1              0            172
Bourne Shell                     3             26             15             90
-------------------------------------------------------------------------------
SUM:                            57           1507           1631          12987
-------------------------------------------------------------------------------


Statistical information for the repository '9b4b94e70467b97ea0b4cd3b' was
gathered on 2022/02/18.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
ISDementyev                     95           486            303            2.60
InfluenceFunctional             30         16125           9003           82.84
Tao Liu                         71          2458           1955           14.55
siminegroup                      1             1              1            0.01

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
ISDementyev                 129           26.5          5.9                4.65
InfluenceFunctional        6252           38.8          4.1                9.77
Tao Liu                    1546           62.9          2.6               23.03
```

**Comment by csoneson** [comment]
@whedon generate pdf from branch JOSS

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.04182/joss.04182/10.21105.joss.04182.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.04182/joss.04182/10.21105.joss.04182.pdf) :page_facing_up: :point_left:

**Comment by taoliu032** [comment]
Thank you all very much! To both reviewers: feel free to ask me any questions about the manuscript and software in any way as you see fit :) 

Wish everyone a wonderful weekend :D

**Comment by rvhonorato** [comment]
This code might have a licensing issue, but since **i am not a licensing expert**, I will raise the point here for discussion:

I have noticed in your installation instructions the following:

> params['ld setup path'], params['ld run path'], etc: paths to various [LightDock](https://lightdock.org/) Python scripts, providing they are in a working directory. No need to change these paths. The paths are used to automatically copy the scripts to ld_scripts/ within a working directory from [lib/lightdock/](https://github.com/InfluenceFunctional/OpenDNA/tree/main/lib/lightdock).

Which then, as expected results in the lightdock scripts being **copied** from the lightdock package and **in** your execution. 

![Screenshot 2022-02-21 at 13 37 19](https://user-images.githubusercontent.com/9445814/154956607-fda88b44-2794-4c58-8530-33999f4eb88c.png)

To my understanding the fact the the script are being **copied** rather than **imported** the lightdock from the library makes necessary for **your** code to also be GPLv3, whereas it is now Apache-2.

Since you are just using the scripts and not modifying them (as far as I know) it is better to import the scripts and use then from within your code to avoid any issues.

Ping brianjimenez (lightdock creator) and also @JoaoRodrigues (the other reviewer) since they might also have some insights into this.

**Comment by brianjimenez** [comment]
Thanks for the ping @rvhonorato 
Indeed, it could be a license conflict here since redistributing code with a different license than LightDock (GPLv3) might not be allowed (but needs a double check for Apache-2 + GPLv3 compatibility). If authors are not modifying the code, simpler could be just to add LightDock as a pip dependency and those scripts would be available in the path.

**Comment by taoliu032** [comment]
@rvhonorato Thank you so much for raising the concern! We are not very experienced in choosing a license for open source software -- I was in fact planning to consult JOSS reviewers on the choice of license :) But certainly we do not want to have any licensing conflict. 

@brianjimenez Thank you for helping out! We got the scripts from https://github.com/lightdock/lightdock/tree/0.9.2/bin and we did include LightDock package using pip when creating a conda virtual environment; Otherwise, those scripts would not run either. 
I will try the direct import for the methods in those scripts. One (probably unnecessary) concern is that: when I looked at the "lightdock" folder under "site-packages" in the conda environment, I did not see those scripts. I will check the LightDock documentation to figure out how to do the import properly. If you can point out any direction or document, it would be super helpful too. Thank you again!

**Comment by brianjimenez** [comment]
Hi @taoliu032 , glad to help! To find LightDock scripts, I assume you use `virtualenv` command (conda should behave in a similar way):

```bash
virtualenv venv
source venv/bin/activate
pip install lightdock
```

After installation ends, you should find LightDock scripts on `venv/bin` folder. Try also `which ant_thony.py` to locate scripts. When the virtual environment (or conda) is activated, all LightDock scripts are available at any path.

Hope it helps!

**Comment by taoliu032** [comment]
@brianjimenez - I removed the commands of copying individual lightdock scripts and our code can still run successfully. Thank you for the suggestion!

@rvhonorato - After removing the script-copying command, I have tested the code both on my local macOS machine and our Linux-based computer cluster (the job waited in the queue for a while). Both tests ran successfully and I have also pushed the updates to the JOSS branch :-) Please feel free to download the new code - sorry for causing any inconvenience! Let me know if there is any other concern or questions, thank you!

**Comment by csoneson** [comment]
👋🏻 Hi all - just wanted to check in how things are going here - let me know if you have any questions!

**Comment by JoaoRodrigues** [comment]
Hi all,

Thanks for the invitation to review and congrats on the submission.

The general idea behind this submission is sound, and follows-up on a 2021 publication from the same authors on E2EDNA v1.0, published in JCIM. From my understanding, the code is essentially a re-write to use OpenMM instead of Tinker as the MD engine. While this is valuable - makes it simpler to install/run - the authors do not realize, in my opinion, this change to its fullest potential. The authors repository is not so much a "package" in the traditional sense, but more of a collection of scripts that automate a certain rigid protocol. I would rather see for instance, NUPACK being an optional dependency - as  a user, I could simply provide my own DNA molecules instead of being forced to use NUPACK. In this sense, I think this repository could use more work to stand out on its own compared to last year's publication.

In addition to this comments, I have a general comment on the repository itself. The authors should take some time to clean up files that are no longer useful for the protocol or that are simply part of the development workflow. Folders named `old`, or IDE config folders (`.idea`) should not be part of a published version of the repository, specially when they are even marked to be ignored in the `.gitignore` file. Same with the existence of both a `requirements.txt` file and an `environment.yml` file, whereas only the latter is used. As such, I believe that the authors should spend some time cleaning up the repository and setting up a more "traditional" structure to help potential users navigate through their code base more easily.

Further, I have a few starter questions about the manuscript, code and, licenses that I think should be clarified. Hopefully these will help the authors improve their work and repository/code.

### Licenses
- You're licensing the tool under the Apache license but you are including data (parameter sets) that falls under a difference license. In particular, I see the parameter files for the Amoeba forcefield taken from Tinker/OpenMM almost verbatim. Did you check with the appropriate developers if this sharing of the forcefield parameter files is allowed under their license, without any attribution?

### Installation
- The installation process is quite complex. As a user, I'd have to register and download NUPACK and MMB, as well as edit a series of files in order to get a functional installation. This is simply a suggestion for the developers to keep in mind.

- Related to the point above, have the authors considered using conda directly to install their software, instead of a custom shell script? `pdbfixer` is available as a conda package, and you could specify pip packages there too, e.g. `lightdock`. The installation could be reduced to a simple: 1) install nupack 2) install mmb 3) run `conda env create -f e2edna-env.yml`. 

- On this last point, the authors should strip the granular version of the env `yaml` file otherwise conda will struggle with versions on anything but the authors' hardware.

- According to the README, the code is only tested on MacOS, although I'd imagine the most use would be on a compute cluster running Linux. Have the authors tried running their code on Linux?

### Misc
- In several sections of their documentation, the authors mention "OpenDNA". Was this the previous name of this package?
- It would be greatly beneficial for a user to have config files with installation paths, simulation settings etc, instead of having to edit source code. Would the authors be open to this change? 

## Comments on the Manuscript
- In the "Statement of Need", the authors mention an "all-python" package several times. Being pedantic, this is not entirely true as their code relies on quite some compiled code in their dependencies (lightdock, openmm).

**Comment by taoliu032** [comment]
@JoaoRodrigues Thank you for the comments! Our responses have been added to the same issue.

**Comment by csoneson** [comment]
👋🏻 @JoaoRodrigues - did you have a chance to take a look at @taoliu032's responses to your review? 
@rvhonorato - could you update us on how your review is progressing? 

Thanks all!

**Comment by csoneson** [comment]
Ping @JoaoRodrigues, @rvhonorato - could you update us on how things are progressing here? Thanks!

**Comment by rvhonorato** [comment]
Sorry for the delay, I'll take a look at this asap!

**Comment by csoneson** [comment]
@JoaoRodrigues, @rvhonorato - any updates here?

**Comment by JoaoRodrigues** [comment]
Hi @csoneson, apologies, I haven't had as much time to work on this as I wanted. Replied to the authors just now. Last thing I would need before 👍 is to test the software locally. Unfortunately, the current examples (AFAIK) require some third-party software that I'd have to register to download, and I don't want to do so. I've asked the authors to provide an example that does not require those modules, since they've mentioned that that is a possibility.

**Comment by taoliu032** [comment]
Thank you for the comment. An example has been prepared as requested, please see the detail here: https://github.com/siminegroup/E2EDNA2/issues/14#issuecomment-1107938433. 

As always, please do not hesitate to ping me should there be any question.

**Comment by rvhonorato** [comment]
@JoaoRodrigues did most of the heavy lifting addressing most of the points I had with the code, thanks ;) For me before the final OK I also just need to test it locally, going over it now based on the link by @taoliu032 above.

**Comment by taoliu032** [comment]
Hi @csoneson, thank you for your patience. I updated the codebase with the changes requested by the reviewers and tested them myself. The reviewers' suggestions are very helpful - thank you very much!

Now we are waiting for any feedback from our beta tester. Once it is done (should not be long), I will update the JOSS branch and invite reviewers to test the code, as the final step hopefully.

Summary of the updates can be seen here: https://github.com/siminegroup/E2EDNA2/issues/14#issuecomment-1124250089.

**Comment by csoneson** [comment]
Great, thanks for the update @taoliu032!

**Comment by JoaoRodrigues** [comment]
👍 for me for publication. Good job with the reviews @taoliu032 and thank you for the patience!

**Comment by csoneson** [comment]
Thanks @JoaoRodrigues and @rvhonorato!

@taoliu032 - I will also have a quick look through the submission and will get back to you with the next steps shortly.

**Comment by taoliu032** [comment]
Thank you very much @JoaoRodrigues @rvhonorato!

@csoneson - sounds good and I have just updated the code in JOSS branch

**Comment by csoneson** [comment]
@editorialbot generate pdf

**Comment by csoneson** [comment]
@editorialbot generate pdf from branch JOSS

**Comment by csoneson** [comment]
@editorialbot set JOSS as branch

**Comment by csoneson** [comment]
@editorialbot generate pdf

**Comment by csoneson** [comment]
@editorialbot check references

**Comment by csoneson** [comment]
@taoliu032 I have made a [PR](https://github.com/siminegroup/E2EDNA2/pull/17) with a couple of minor edits to the manuscript - please check and merge if you agree. After merging, you can generate a new proof with `@editorialbot generate pdf` as above. 

The next steps then are as follows: 
- [ ] Make a tagged release of the software, and list the version tag of the archived version here.
- [ ] Archive the reviewed software in Zenodo or a similar service
- [ ] Check that the archival deposit (e.g., in Zenodo) has the correct metadata. In particular, the title and author list should be identical to those of the paper. You can also add the authors' ORCID.
- [ ] Please list the DOI of the archived version here.

I can then move forward with accepting the submission.

**Comment by taoliu032** [comment]
@csoneson Thank you very much! I have merged the commits and I will work on the list of tasks. Have a good weekend.

**Comment by taoliu032** [comment]
@editorialbot generate pdf

**Comment by taoliu032** [comment]
@editorialbot check references

**Comment by taoliu032** [comment]
Hi @csoneson! I believe I have completed the following steps.
> * [x]  Make a tagged release of the software, and list the version tag of the archived version here.

The release is [here](https://github.com/siminegroup/E2EDNA2/releases/tag/v2.0.0) and the version tag is `v2.0.0`, in order to be explicitly distinguished from the original version of E2EDNA.

We would like to ask a question regarding the version: When we made our initial submission to JOSS, we were asked to provide a "Software version". We first tried "v2.0.0" but the submission portal did not allow us to do so, probably because this is our very first paper in JOSS about this software and we chose "New submission" as "Type of submission". So we ended up putting down "v1.0", which is now shown at the top of this review issue and quoted below:
>> **Submitting author:** @taoliu032 ([Tao Liu](http://orcid.org/0000-0002-1082-5570))
> **Repository:** https://github.com/siminegroup/E2EDNA2
> **Branch with paper.md** (empty if default branch): JOSS
> **Version:** v1.0

Is it okay to have such a discrepancy, submission version "v1.0" v.s. release version "v2.0.0"? Where would the "v1.0" and "v2.0.0" show up later?

> * [x]  Archive the reviewed software in Zenodo or a similar service
> * [x]  Check that the archival deposit (e.g., in Zenodo) has the correct metadata. In particular, the title and author list should be identical to those of the paper. You can also add the authors' ORCID.

The deposit was created in Zenodo and metadata was checked: https://zenodo.org/record/6546661

> * [x]  Please list the DOI of the archived version here.

DOI on Zenodo: 10.5281/zenodo.6546661

Thank you again!

**Comment by csoneson** [comment]
Great, thanks @taoliu032! I will update the version, and it should be correct in the published version.

**Comment by csoneson** [comment]
@editorialbot set v2.0.0 as version

**Comment by csoneson** [comment]
@editorialbot set 10.5281/zenodo.6546661 as archive

**Comment by csoneson** [comment]
@taoliu032 I will now hand over to the associate editor-in-chief on rotation for the last steps. Thanks for submitting to JOSS!

**Comment by csoneson** [comment]
@editorialbot recommend-accept

**Comment by taoliu032** [comment]
@csoneson Thank you for all the help! Let me know if there is anything we can do to help to resolve the error above.

**Comment by arfon** [comment]
/ cc @tarleb – any thoughts on how we should handle this? It looks like the BibTeX file has two ISSNs.

**Comment by tarleb** [comment]
That's interesting. I'm looking into it now.

**Comment by tarleb** [comment]
I've pushed a change, now only the first ISSN will be used.

Publications can have an ISSN for their print version and another for the electronic (online) version. The latter is sometimes called the eISSN and it seems that Zotero adds both.

There may be other ways to improve our handling of the ISSN field, but it should work for now.

**Comment by arfon** [comment]
:zap: thanks @tarleb !

**Comment by taoliu032** [comment]
Hi @arfon! Thanks for getting us help to solve the ISSN issue. May I ask what steps are left for accepting the manuscript? Our editor recommended acceptance but the command to the robot may not completely go through, due to the ISSN issue.

**Comment by csoneson** [comment]
@editorialbot recommend-accept

**Comment by tarleb** [comment]
Huh :confused: 

Sorry about that, I'll look into it again.

**Comment by csoneson** [comment]
Thanks @tarleb! Looks like the same fix may need to be applied to the ISBN field 🙂

**Comment by tarleb** [comment]
OK, applied the fix to that field as well. Should work now. :crossed_fingers:

**Comment by csoneson** [comment]
@editorialbot recommend-accept

**Comment by csoneson** [comment]
Yay! Thanks for the fix @tarleb

**Comment by csoneson** [comment]
@taoliu032 This will now go to the associate editor-in-chief on rotation for the last steps. Thanks for your patience!

**Comment by taoliu032** [comment]
Hooray! Thank you @csoneson and thank you for the fix @tarleb

**Comment by danielskatz** [comment]
@taoliu032 - I'm the AEiC this week, and have a few small changes for the paper: please merge https://github.com/siminegroup/E2EDNA2/pull/20 or let me know what you disagree with, then we can proceed to publication.

**Comment by danielskatz** [comment]
@editorialbot recommend-accept

**Comment by danielskatz** [comment]
@editorialbot accept

**Comment by danielskatz** [comment]
Congratulations to @taoliu032 (Tao Liu) and co-authors!!

And thanks to @JoaoRodrigues and @rvhonorato for reviewing, and @csoneson for editing!
We couldn't do this without you

**Comment by taoliu032** [comment]
Thank you to all of you for your time and help!
